docs/running-clusterfuzzlite/gitlab.md

```diff
@@ -0,0 +1,229 @@
+---
+layout: default
+parent: Step 2&#58; Running ClusterFuzzLite
+grand_parent: ClusterFuzzLite
+title: GitLab
+nav_order: 1
+permalink: /running-clusterfuzzlite/gitlab/
+---
+# GitLab
+{: .no_toc}
+
+- TOC
+{:toc}
+---
+
+This page explains how to set up ClusterFuzzLite to run on [GitLab].
+To get the most of this page, you should have already set up your
+[build integration] and read the more
+[high-level document on running ClusterFuzzLite].
+
+## Gitlab runner
+The following examples use a `docker` gitlab runner running sibling containers:
+[doc](https://docs.gitlab.com/ee/ci/docker/using_docker_build.html#use-docker-socket-binding).
+
+It should be possible to achieve the same functionalities with using a shell executor.
+But then, the `.gitlab-ci.yml` should be different, and explicitly call the `docker` commands
+on clusterfuzz-lite images.
+
+Docker-in-Docker does not seem possible as clusterfuzz-lite images
+do not have the docker tools installed.
+
+## .gitlab-ci.yml
+For basic ClusterFuzzLite functionality, all you need is a single job
+to enable fuzzing on your merge requests.
+
+To enable more features, we recommend having different jobs for:
+
+- continuous builds
+- batch fuzzing and corpus pruning
+- coverage
+
+### MR fuzzing
+
+To add a fuzzing job that fuzzes all merge requests to your repo, add the
+following default configurations to `.gitlab-ci.yml`:
+
+{% raw %}
+```yaml
+variables:
+  SANITIZER: "address"
+  CFL_PLATFORM: gitlab
+
+clusterfuzzlite:
+  image:
+    name: google/clusterfuzzlite/actions/build_fuzzers@v1
+    entrypoint: [""]
+  stage: fuzz
+  rules:
+    # default code change
+    - if: $CI_PIPELINE_SOURCE == "merge_request_event"
+      variables:
+        MODE: "code-change"
+  before_script:
+    # gitlab's container name is not its hostname !
+    - export CFL_CONTAINER_ID=`cut -c9- < /proc/1/cpuset`
+  script:
+    # will build and run the fuzzers
+    - python3 "/opt/oss-fuzz/infra/cifuzz/cifuzz_combined_entrypoint.py"
+  artifacts:
+    # upload artifacts when a crash makes the job fail
```

Comment:
`always` means `always but when there is something to upload`
It seems simpler for me : this way we use consistently `always` for all the builds.
For batch fuzzing, we need always : because we need crashes on failures, and corpus on success